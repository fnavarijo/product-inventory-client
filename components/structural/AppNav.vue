<template>
  <nav class="absolute top-0 right-0 h-32 flex items-center p-16">
    <!-- <label for="username">Username</label>
    <input type="text" v-model="credentials.username" name="username">
    <label for="password">Password</label>
    <input type="text" v-model="credentials.password" name="password">
    <button @click="doLogin" class="bg-gray-500">Login</button> -->
    <!-- TODO: On hover, paint path white -->
    <NuxtLink to="/login">
      <img class="nav-icon" src="~/assets/img/user.svg" alt="Go to shopping cart">
    </NuxtLink>
    <img class="nav-icon" src="~/assets/img/shopping.svg" alt="Go to shopping cart">
  </nav>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  // data () {
  //   return {
  //     credentials: {
  //       username: '',
  //       password: '',
  //     },
  //   },
  // },
  // methods: {
  //   async doLogin () {
  //     const { username, password } = this.credentials;
  //     // TODO: move this to a centralized api management
  //     const { data } = await this.$axios.post('/auth/local', {
  //       identifier: username,
  //       password,
  //     });

  //     localStorage.setItem('session', data.jwt);
  //     this.$router.push('/home');

  //   }
  // }
});
</script>

<style scoped>
.nav-icon {
  @apply h-10 w-10 m-5;
}
</style>
