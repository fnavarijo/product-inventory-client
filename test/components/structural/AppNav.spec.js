import { mount, RouterLinkStub } from '@vue/test-utils';
import AppNav from '@/components/structural/AppNav.vue';

describe('AppNav Component', () => {
  it('should have User and Cart buttons', () => {
    const wrapper = mount(AppNav, {
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
