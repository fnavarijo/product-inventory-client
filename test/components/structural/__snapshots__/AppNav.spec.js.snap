// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav Component should have User and Cart buttons 1`] = `<nav class="absolute top-0 right-0 h-32 flex items-center p-16"><a><img src="~/assets/img/user.svg" alt="Go to shopping cart" class="nav-icon"></a> <img src="~/assets/img/shopping.svg" alt="Go to shopping cart" class="nav-icon"></nav>`;
